pkg/controllers/job/state/running.go

```diff
@@ -56,6 +56,13 @@ func (ps *runningState) Execute(action v1alpha1.Action) error {
 				// when scale down to zero, keep the current job phase
 				return false
 			}
+
+			minSuccess := ps.job.Job.Spec.MinSuccess
+			if minSuccess != nil && status.Succeeded >= *minSuccess {
+				status.State.Phase = vcbatch.Completed
+				return true
+			}
+
 			if status.Succeeded+status.Failed == jobReplicas {
 				if status.Succeeded >= ps.job.Job.Spec.MinAvailable {
 					status.State.Phase = vcbatch.Completed
```

Comment:
If we use `minSuccess`, do we still need `MinAvailable` to determine job status?

Reply:
We can use `MinAvailable` to determine job status when users don't want to use `MinSuccess`

Reply:
yes, l agree. So maybe we can add another layer of logic inside `if status.Succeeded+status.Failed == jobReplicas ` ? If `minSuccess` is specified, we use `minSuccess` instead of `minAvailable`

Reply:
Yes

Reply:
If `minSuccess` has matched，Job will mark `Completed`. So it's not necessary to handle the next compare logic.

Reply:
l mean if `minSuccess` is specified while `status.Succeeded < *minSuccess` all the time, then after all the pods completed, we should also replace `if status.Succeeded >= ps.job.Job.Spec.MinAvailable` with `if status.Succeeded >= minSuccess`. WDYT?

Reply:
Agree